### PR TITLE
docs: add VueSchool Summer Sale banner

### PR DESCRIPTION
**Pull request by nicodevs**
This PR adds a banner on top of router.vuejs.org to inform users about Vue School’s Summer Sale promotion.

The banner is closable and opens Vue School's Summer Sale promo page in a new tab. The banner will not reappear if a user has closed it (stored in local storage).

The banner is based on the one used on vuejs.org.

[Click here to see screenshots of the banner](https://postimg.cc/gallery/gLTRndG)

### CODE REVIEW

File: docs/.vuepress/theme/Layout.vue [hunk 1]
```diff
@@ -1,61 +1,82 @@
...
+      </template>
+    </ParentLayout>
+  </div>
 </template>
 
 <script>
 import ParentLayout from '@parent-theme/layouts/Layout.vue'
 import CarbonAds from './components/CarbonAds.vue'
 import BuySellAds from './components/BuySellAds.vue'
+import BannerTop from './components/BannerTop.vue'
```
Comment (posva) [inline docs/.vuepress/theme/Layout.vue]: can we make it async as well?

**Review by posva — CHANGES_REQUESTED** [review]

File: docs/.vuepress/theme/Layout.vue (hunk 1 above)
Comment (nicodevs) [reply docs/.vuepress/theme/Layout.vue]: Done! Let me know if it's OK @posva

File: docs/.vuepress/components/HomeSponsors.vue [hunk 2]
```diff
@@ -27,11 +27,17 @@ export default {
...
-  padding: 35px 40px 45px;
-  margin: 0 -2rem;
+  padding: 35px 30px 45px;
   background-color: #f6f6f6;
 }
 
+@media (min-width: 768px) {
+  #sponsors {
+    padding: 35px 40px 45px;
+    margin: 0 -2rem;
```
Comment (posva) [inline docs/.vuepress/components/HomeSponsors.vue:37]: why this change? it changed the look on mobile
Screen Shot 2021-06-10 at 09 56 06

Is this change by mistake?

Comment (nicodevs) [reply docs/.vuepress/components/HomeSponsors.vue:37]: I removed the negative left and right margin because it was generating:
- an horizontal scrollbar on mobile
- and a margin issue on the top navbar (that is present on the production website, but it became even more pronounced when adding the banner).

To replicate the issue, visit the site on mobile and scroll down. You'll see that the top margin of the navbar gets collapsed. And scroll to the sides, you will see the website has an horizontal scrollbar.

Here's a screenshot: https://imgur.com/a/v7jMwEY

Feel free to ignore this fix if you don't think is appropriate. Doing so will cut some pixels from the top of the banner.

Let me know if you have questions.

Comment (posva) [reply docs/.vuepress/components/HomeSponsors.vue:37]: I see, thanks, let's keep it then!
